presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/execution/thrift/ThriftSqlFunctionExecutor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.execution.thrift;
+
+import com.facebook.drift.TException;
+import com.facebook.drift.client.DriftClient;
+import com.facebook.presto.common.NotSupportedException;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.TypeSignature;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.SqlFunctionHandle;
+import com.facebook.presto.spi.function.SqlFunctionId;
+import com.facebook.presto.spi.function.ThriftScalarFunctionImplementation;
+import com.facebook.presto.thrift.api.datatypes.PrestoThriftBlock;
+import com.facebook.presto.thrift.api.udf.ThriftFunctionHandle;
+import com.facebook.presto.thrift.api.udf.ThriftUdfService;
+import com.facebook.presto.thrift.api.udf.ThriftUdfServiceException;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static com.facebook.airlift.concurrent.MoreFutures.toCompletableFuture;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.facebook.presto.thrift.api.udf.ThriftUdfPage.thriftPage;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class ThriftSqlFunctionExecutor
+{
+    private final DriftClient<ThriftUdfService> thriftUdfClient;
+
+    @Inject
+    public ThriftSqlFunctionExecutor(DriftClient<ThriftUdfService> thriftUdfClient)
+    {
+        this.thriftUdfClient = thriftUdfClient;
```

Comment:
`requireNonNull`.
- Guice usually doesn't inject nulls. If binding does not exists, it simply throws.
- Because you're binding this DriftClient and this ThriftSqlFunctionExecutor at the same time in `SimpleAddressThriftSqlFunctionExecutionModule`, DriftClient always exists.